openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -25,6 +36,23 @@ func (e *extension) Transformers(v1alpha1.KComponent) []mf.Transformer {
 func (e *extension) Reconcile(ctx context.Context, comp v1alpha1.KComponent) error {
 	ks := comp.(*v1alpha1.KnativeServing)
 
+	// Fetch the proper domain.
+	ingress, err := e.ocpclient.ConfigV1().Ingresses().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch cluster config: %w", err)
+	}
+	domain := ingress.Spec.Domain
+	if domain != "" {
+		common.Configure(&ks.Spec.CommonSpec, "domain", domain, "")
+	}
+
+	// Attempt to locate kibana route which is available if openshift-logging has been configured
+	route, err := e.ocpclient.RouteV1().Routes("openshift-logging").Get("kibana", metav1.GetOptions{})
+	if err == nil && len(route.Status.Ingress) > 0 && route.Status.Ingress[0].Host != "" {
+		common.Configure(&ks.Spec.CommonSpec, "observability", "logging.revision-url-template",
+			fmt.Sprintf(loggingURLTemplate, route.Status.Ingress[0].Host))
+	}
```

Comment:
I think this looks more compatc that on the `serving.go` file on the `knative-operator`.

Reply:
Refactored into a function for clarity.